automerge-frontend/src/state_tree/resolved_path.rs

```diff
@@ -284,161 +234,133 @@ impl ResolvedCounter {
     }
 }
 
-pub struct ResolvedMap {
-    pub(super) value: StateTreeMap,
-    pub(super) multivalue: MultiValue,
-    pub(super) focus: Focus,
+pub struct ResolvedMap<'a> {
+    pub(super) multivalue: &'a mut MultiValue,
 }
 
-impl ResolvedMap {
+impl<'a> ResolvedMap<'a> {
     pub(crate) fn set_key(
-        &self,
+        &mut self,
         key: &str,
         payload: SetOrInsertPayload<&Value>,
     ) -> LocalOperationResult {
+        let state_tree_map = match self.multivalue.default_statetree_value_mut() {
+            StateTreeValue::Composite(StateTreeComposite::Map(map)) => map,
+            _ => unreachable!(),
```

Comment:
Is it possible to avoid these `unreachable!` assertions by giving the `StateTreeComposite::*` to the `Resolved*` directly so that we don't need the `match`?

Reply:
I'm not sure we can, we need to store the `MultiValue` so we can get the conflicts. But then we can't also have a mutable reference to the composite.